### ISSUE TITLE
Stream publishing from publisher's side but not playing in player side

### ISSUE THREAD

**Issue by cloud-github** [ant-media/Ant-Media-Server#3060]
### Short description
I am using two different port for playing (5443) and publishing(443) for implementing conference using ant version 2.2.1
Sometime there is a problem with roomInformation data in playing side. Publisher is able to publish stream correctly and can be tallied on ant CMS as well. But on the other hand the published stream is not available in roomInformation in playing side.

### Environment

* Operating system and version: ALL OS
* Ant Media Server version: Enterprise Edition 2.2.1
* Browser name and version: ALL Browser

### Steps to reproduce
1.  Arrange a conference call with 5 - 6 people for around 30 - 40 minutes using conference.html
2.  If possible stay in weak network so that system will try to reconnect again in that room
3.  Let some user leave and rejoin the room

### Expected behavior
After each leave and join operation, users must watch and be watchable everyone.  
### Actual behavior
Some users leave and join but not all people are watchable.

**Comment by golgetahir** [comment]
Hi,

Assuming you are using the conference.html, there is no mechanism for it to republish if required. Join and publish commands are different in conference room in the Ant Media architecture, you can join a conference room without publishing which may be what you see.

Republish method is there in index.html, you can implement something like that in the conference.html also any commits are welcomed if you want to share it with us.

```
function checkAndRepublishIfRequired() {
	 	var iceState = webRTCAdaptor.iceConnectionState(streamId);
		console.log("Ice state checked = " + iceState);

	  	if (iceState == null || iceState == "failed" || iceState == "disconnected"){
	  		webRTCAdaptor.stop(streamId);
	  		webRTCAdaptor.closePeerConnection(streamId);
	  		webRTCAdaptor.closeWebSocket();
	  		initWebRTCAdaptor(true, autoRepublishEnabled);
		  }	
	}
```


Cheers

**Comment by mekya** [comment]
Another reproduce scenario is as follows

1. Join the conference.html with five people. 
2. Play it for 30 mins 
3. After that, let some users leave and join the room again. 

Expected behaviour:
After each leave and join operation, users must watch and be watchable everyone.  

Actual behaviour: 
Some users leave and join but not all people are watchable.

**Comment by eneskuluk** [comment]
Let me write my findings on this issue. I have tested on the sample conference.html in a weak network conditions.
What i have found is in weak network conditions, sometimes outside connections may cut and play_finished may happen.
If an incoming stream that its play has finished is still streaming which means it is still in the room, wouldn't be played since it is already in the streamsList. I have fixed it by removing it from streamsList. So, when roominformation call invoked, it will play the finished stream since it is not in the streamsList anymore.

On the other hand, when publish_finished happened, that specific stream is removed from the room (also in the roomInformation) and did not appear in the roominformation callbacks. I think this is expected outcome. But it kept playing other streams since its websocket connection did not disconnected and was still getting roomInformation.

I sent a fix that might help in this situation but i haven't able to reproduce the issue you got which is missing stream ids' in the  roomInformation callback.

What I tried:
* 5 people with a weak network connection in conference room, sometimes publish and timeout happened and reconnected back but no issue in the roominformation callback. The stream that it got timeout was not in the roominformation as expected.
